pkg/terraform/stages/gcp/stages.go

```diff
@@ -42,3 +53,95 @@ func removeFromLoadBalancers(s stages.SplitStage, directory string, terraformDir
 		"failed disabling bootstrap load balancing",
 	)
 }
+
+// extractGCPLBConfig extracts the load balancer information from the terraform outputs, generates the
+// Load Balancer Config file, regenerates the bootstrap ignition, and updates the terraform variables file.
+func extractGCPLBConfig(s stages.SplitStage, directory string, terraformDir string, file *asset.File, tfvarsFile *asset.File) (string, error) {
+
+	// Convert the terraform outputs file into json to extract LB data
+	outputs := map[string]interface{}{}
+	err := json.Unmarshal(file.Data, &outputs)
+	if err != nil {
+		return "", err
+	}
+
+	userConfiguredDNSRaw, ok := outputs["user_configured_dns"]
+	if !ok {
+		return "", fmt.Errorf("failed to read user configured dns from terraform inputs")
+	}
+	if !userConfiguredDNSRaw.(bool) {
+		return "", nil
+	}
+
+	// Extract the Load Balancer ip addresses from the terraform output.
+	apiLBIpRaw, ok := outputs["cluster_public_ip"]
+	if !ok {
+		return "", fmt.Errorf("failed to read External API LB DNS Name from terraform outputs")
+	}
+	apiIntLBIpRaw, ok := outputs["cluster_ip"]
+	if !ok {
+		return "", fmt.Errorf("failed to read Internal API LB DNS Name from terraform outputs")
+	}
+
+	// Create the load balancer configuration file, and add the parsed terraform variables
+	// from above.
+	lbConfigContents, err := lbconfig.CreateLBConfigMap("openshift-lbconfig-custom-dns", apiIntLBIpRaw.(string), apiLBIpRaw.(string))
+	if err != nil {
+		return "", fmt.Errorf("failed to create load balancer config contents: %w", err)
+	}
+	pth := fmt.Sprintf("%s/%s", directory, lbconfig.ConfigName)
+	if err := os.WriteFile(pth, []byte(lbConfigContents), 0o640); err != nil {
+		return "", fmt.Errorf("failed to rewrite %s: %w", lbconfig.ConfigName, err)
+	}
+
+	// Assets to be loaded, edited/updated, and regenerated. The bootstrap asset causes the ignition to be
+	// regenerated. The regenerated certs will be used during ignition.
+	assetsToRegen := []asset.WritableAsset{
+		&bootstrap.Bootstrap{},
+		&tls.KubeAPIServerExternalLBServerCertKey{},
+		&tls.KubeAPIServerInternalLBServerCertKey{},
+		&tls.KubeAPIServerServiceNetworkServerCertKey{},
+	}
+
+	store, err := assetstore.NewStore(directory)
+	if err != nil {
+		return "", fmt.Errorf("failed to create asset store: %w", err)
+	}
+	for _, a := range assetsToRegen {
+		if err := store.Destroy(a); err != nil {
```

Comment:
If you destroy the asset here it's going to delete the bootstrap ignition from disk. So any user changes to that ignition are going to be lost.

Reply:
The only way to regenerate the asset and pick up new changes is to destroy it. If we don't, it is just reloaded from the state file. I see the concern, but is there a way around it ?

Reply:
Regenerating the asset goes against the intended design of the asset graph, so I don't think we should be trying work around that. For the specific goal of editing bootstrap ignition after it's generated, editing in memory would be less problematic, but is far from ideal.